src/v/raft/consensus.cc

```diff
@@ -2129,6 +2129,15 @@ consensus::do_append_entries(append_entries_request&& r) {
         reply.may_recover = _follower_recovery_state->is_active();
     }
 
+    // abort here to pretend we are in recovery if we would without abortion
+    if (unlikely(_append_entries_blocked)) {
+        vlog(
+          _ctxlog.warn,
+          "ignoring append_entries request from {}: blocked",
+          r.source_node());
+        co_return reply;
+    }
```

Comment:
Why are we blocking in this particular place and not earlier, i would imagine a failure injection is a general mechanism that prevents the follower from receiving append entries. Here we add a context to it as it is in a particullar place. I think we should reconisder the way this failure injection is done and try to approch it with more generic mechanism rather than adding a single flag